controllers/articles.js

```diff
@@ -11,13 +11,24 @@ function getArticles(req,res,next) {
 
 function getArticle(req,res,next) {
   const {article_id} = req.params
-  return findArticle(article_id).then((result)=>{
-    const article = result[0]
+  return findArticle(article_id).then((article)=>{
     res.status(200).send({article})
   })
   .catch((err)=>{
     next(err)
   })
 }
 
-module.exports = {getArticle, getArticles}
\ No newline at end of file
+function patchArticle(req,res,next) {
+  const {article_id} = req.params
+  const {inc_votes} = req.body
+  Promise.all([updateArticle(article_id, inc_votes), checkArticleExists(article_id)])
```

Comment:
We don't necessarily need a checkExists function here, we only needed that previously when we couldn't rely on the response from the database query to determine whether it was an error or not, in this scenario see what happens if we process a query through psql where the article_id is not found we may be able to handle it that way 👍

Reply:
This is now handled in the model, with a logic statement that rejects the Promise chain if rows is empty.